### ISSUE TITLE
Raster DisplayList::Dispatch take time too long

### ISSUE THREAD

**Issue by AbeiOS** [flutter/flutter#140727]
### Steps to reproduce

1. Use CupertinoDatePicker.
2. Quickly Scroll down and up a item some minute.
3. Look Performance.
4. DisplayList::Dispatch take 50ms.
![FE2C776E-D8D4-4D8B-AC5B-13275CA63510](https://github.com/flutter/flutter/assets/12441461/21090983-1e0f-4eb4-a6e5-5b79b65f5865)

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  DateTime initDate = DateTime(
      DateTime
          .now()
          .year,
      DateTime
          .now()
          .month,
      DateTime
          .now()
          .day,
      0,
      0,
      0,
      0,
      0);

  MyHomePage({Key? key, required this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      // This call to setState tells the Flutter framework that something has
      // changed in this State, which causes it to rerun the build method below
      // so that the display can reflect the updated values. If we changed
      // _counter without calling setState(), then the build method would not be
      // called again, and so nothing would appear to happen.
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Scaffold(
      appBar: AppBar(
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: Text(widget.title),
      ),
      body: Center(
        // Center is a layout widget. It takes a single child and positions it
        // in the middle of the parent.
        child: Column(
          // Column is also a layout widget. It takes a list of children and
          // arranges them vertically. By default, it sizes itself to fit its
          // children horizontally, and tries to be as tall as its parent.
          //
          // Invoke "debug painting" (press "p" in the console, choose the
          // "Toggle Debug Paint" action from the Flutter Inspector in Android
          // Studio, or the "Toggle Debug Paint" command in Visual Studio Code)
          // to see the wireframe for each widget.
          //
          // Column has various properties to control how it sizes itself and
          // how it positions its children. Here we use mainAxisAlignment to
          // center the children vertically; the main axis here is the vertical
          // axis because Columns are vertical (the cross axis would be
          // horizontal).
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            SizedBox(
              width: 300.0,
              height: 300.0,
              child: CupertinoDatePicker(
                mode: CupertinoDatePickerMode.date,

                onDateTimeChanged: (value) {
                  widget.initDate = DateTime(
                      value.year,
                      value.month,
                      value.day,
                      0,
                      0,
                      0,
                      0,
                      0);
                },
                minimumYear: DateTime
                    .now()
                    .year - 5,
                maximumYear: DateTime
                    .now()
                    .year,
                maximumDate: DateTime.now(),
                initialDateTime: DateTime(widget.initDate.year, widget.initDate.month, widget.initDate.day),

              ),),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}

```

</details>

### What target platforms are you seeing this bug on?

Android

### OS/Browser name and version | Device information

Any

### Does the problem occur on emulator/simulator as well as on physical devices?

Unknown

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[compile]: command =  [/Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java, -version]
[compile]: command =  [/Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java, -version]
[compile]: command =  [which, pod]
[compile]: command =  [/Users/luffy/Library/Android/sdk/platform-tools/adb, devices, -l]
[compile]: command =  [/usr/bin/arch, -arm64e, xcrun, xcdevice, list, --timeout, 2]
[compile]: command =  [/usr/bin/arch, -arm64e, xcrun, simctl, list, --json, devices]
[✓] Flutter (Channel feature/refactor/koi, 3.0.0, on macOS 13.3.1 22E772610a darwin-arm (Rosetta), locale zh-Hans-CN)
    • Flutter version 3.0.0 at /Users/luffy/zcy_flutter_sdk_3_0_0
    • Upstream repository https://git.cai-inc.com/mobile-framework/flutter-components/flutter_sdk_3_0_0.git
    • Framework revision 9cd167ad8f (3 days ago), 2023-12-26 21:14:16 +0800
    • Engine revision d1b9a6938a
    • Dart version 2.17.0
    • DevTools version 2.12.2
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

⣽[compile]: command =  [pod, --version]
⣟[compile]: command =  [/Users/luffy/Library/Android/sdk/platform-tools/adb, -s, 7HX5T19A11005536, shell, getprop]
⡿[compile]: command =  [/Applications/Google Chrome.app/Contents/MacOS/Google Chrome, --version]
⣿[compile]: command =  [/Users/luffy/Library/Android/sdk/platform-tools/adb, devices, -l]
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/luffy/Library/Android/sdk
    • Platform android-33, build-tools 30.0.3
    • ANDROID_HOME = /Users/luffy/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2022.1.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.85.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
    • LIO AL00 (mobile) • 7HX5T19A11005536 • android-arm64  • Android 12 (API 31)
    • macOS (desktop)   • macos            • darwin-arm64   • macOS 13.3.1 22E772610a darwin-arm (Rosetta)
    • Chrome (web)      • chrome           • web-javascript • Google Chrome 120.0.6099.129

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
(base)  luffy@MacBook-Pro-9  ~/FlutterProject/zcy_app  feature/refactor/koi ● 
```

</details>

**Comment by darshankawar** [comment]
@AbeiOS 
There was a similar [issue](https://github.com/flutter/flutter/issues/130141) fixed recently that you can check and verify at your end. The fix https://github.com/flutter/flutter/issues/132416 should be available in latest stable version or in master.

**Comment by AbeiOS** [comment]
> @AbeiOS There was a similar [issue](https://github.com/flutter/flutter/issues/130141) fixed recently that you can check and verify at your end. The fix #132416 should be available in latest stable version or in master.

Ok, I will have try.
